dev/PushNotifications/PushNotificationManager.cpp

```diff
@@ -156,7 +156,20 @@ namespace winrt::Microsoft::Windows::PushNotifications::implementation
                         THROW_IF_FAILED(CreateChannelWithRemoteIdHelper(appUserModelId, remoteId, channelInfo));
 
                         // Register a sink with platform which is initialized in the current process
-                        THROW_IF_FAILED(PushNotifications_RegisterNotificationSinkForFullTrustApplication(appUserModelId.get(), PushNotificationManager::Default().as<ABI::Microsoft::Internal::PushNotifications::INotificationListener>().get()));
+                        bool registeredEvent{ false };
+                        {
+                            auto lock{ m_lock.lock_shared() };
+                            registeredEvent = (bool) m_foregroundHandlers;
+                        }
+
+                        if (registeredEvent)
+                        {
+                            {
+                                auto lock{ m_lock.lock_exclusive() };
+                                m_sinkRegisteredWithPlatform = true;
```

Comment:
What is the rationale for this bool?

Reply:
So that we only register a sink if the user defined a foreground handler. Any other activation should trigger background activation.

Reply:
We only want to register a sink if there are foreground handlers registered. Otherwise we want background activation to handle delivering payload.